ESLint: Honor .eslintignore if present

brgaulin opened the pull request:
Hello Hound Team =)

This is a first pass at abstracting .jshintignore logic so eslint can use the same stuff. I always welcome feedback on anything!

This fixes #1071

brgaulin commented on app/models/linter/jshint.rb:
```diff
@@ -1,11 +1,7 @@
-        File.fnmatch?(pattern, commit_file.filename)
-      end
-    end
+    FILE_REGEXP = /.+\.js\z/
```
this is all set now =) it was really the line above it causing problems

brgaulin commented on spec/models/linter/eslint_spec.rb:
```diff
@@ -67,10 +67,51 @@
+        expect(linter.file_included?(commit_file2)).to be false
+      end
+    end
+  end
```
This is all set now =) had to pull the describe over

gylaz commented on app/models/config/js_ignore.rb:
```diff
@@ -0,0 +1,17 @@
+module Config
+  module JsIgnore
```
What do you think about making `JsIgnore` be a class that Javascript linters use? Composition over inheritance.

gylaz commented on app/models/config/js_ignore.rb:
```diff
@@ -0,0 +1,17 @@
+    def ignored_paths
+      ignore_file = hound_config.content.
+                    fetch("javascript", {}).
+                    fetch("ignore_file", ignore_filename)
```
Our styleguides prefer shallow nesting:
```rb
ignore_file = hound_config.
  content.
  fetch("javascript", {}).
  fetch("ignore_file", ignore_filename)
```

brgaulin replied:
sure! I think rubocop actually yelled at me though =( how can I run the styles locally? what I had done was rubocop --style=.ruby-style <filename>

gylaz commented on app/models/config/jshint.rb:
```diff
@@ -1,12 +1,6 @@
-        load_javascript_ignore.presence || DEFAULT_EXCLUDED_FILES
-      )
-    end
+    include JsIgnore
```
Perhaps this class passes `hound_config` and `ignore_filename` when it initializes the `JsIgnore` object (see my suggestion above)?

brgaulin replied:
I did this originally, I think I do like that way better. The reason I didn't end up doing that was because then that js_ignore class shares concerns of config and linter, but its not that much overlap, it really is all config

gylaz commented on spec/models/config/js_ignore_spec.rb:
```diff
@@ -0,0 +1,49 @@
+  end
+
+  def stubbed_config(commit)
+    Config::Jshint.new(stubbed_hound_config(commit), "jshint")
```
Feels like we're testing `Jshint` here rather than `JsIgnore`.

brgaulin replied:
ya, this was intentional for now (until the PR got picked apart =P)

sometimes I find testing by example a bit easier, but I do plan to refactor this with the above changes =)

gylaz commented on spec/models/linter/js_ignore_spec.rb:
```diff
@@ -0,0 +1,52 @@
+  end
+
+  def stub_config(excluded_paths)
+    linter.config = Struct.new(:excluded_paths).new(excluded_paths)
```
What if this method was called `stub_linter` and returned the linter? Then we could get rid of the `let(:linter)` mystery guest.

brgaulin replied:
I will take a look into this =) Thanks!

gylaz commented:
@brgaulin Had a some feedback/suggestions. Nice work!

brgaulin commented:
Greg -

These are all great suggestions. I will try and update my pr tomorrow.
On Tue, Mar 15, 2016 at 10:55 PM Greg Lazarev <notifications@github.com>
wrote:

> @brgaulin <https://github.com/brgaulin> Had a some feedback/suggestions.
> Nice work!
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/houndci/hound/pull/1104#issuecomment-197127647>
>
-- 
--
Brendan Gaulin

brgaulin commented on spec/support/helpers/linter_helper.rb:
```diff
@@ -1,7 +1,8 @@
 module LinterHelper
-  def build_linter(build = build(:build))
+  def build_linter(build = build(:build), extra_files = {})
```
not sure what to do about this one

brgaulin commented:
@gylaz I've updated the PR and corrected the violations, was wondering if you had any more thoughts(sorry for so many style violations, I guess I'm set in my ways =P)

brgaulin commented:
circle did pass, I'm moving the rubocop stuff to its own PR

teoljungberg commented:
I've merged this as a162e36. I modified the code to fit in better with our guidelines, like avoiding using `let`. 

Thanks for your contribution, @brgaulin! This is deployed to production now. Do you mind testing it out to make sure it all works as you intended?

brgaulin commented:
@teoljundberg not sure how I didn't link this, but this PR is also required to enable functionality. 

https://github.com/houndci/eslint/pull/22

brgaulin commented:
@teoljungberg ^ 
GitHub mobile is being a pain for me lol

teoljungberg commented:
@brgaulin I think you are mixing your PRs together. This has only to with ignoring files with the `.eslintignore` file.

brgaulin commented:
I am :) you are correct.
On Tue, Mar 29, 2016 at 9:37 AM Teo Ljungberg <notifications@github.com>
wrote:

> @brgaulin <https://github.com/brgaulin> I think you are mixing your PRs
> together. This has only to with ignoring files with the .eslintignore
> file.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/houndci/hound/pull/1104#issuecomment-202897363>
>
-- 
--
Brendan Gaulin
